File: modules/python_workloads/src/python_workloads/workload_n.py
"""Workload sizing shared by the bench and memory targets."""

import os


def workload_n(fallback: int) -> int:
    """Workload size from WORKLOAD_N, falling back to the target's default."""
    raw = os.environ.get("WORKLOAD_N")
    if raw is None:
        return fallback
    try:
        return int(raw)
    except ValueError:
        return fallback

File: modules/java_workloads/src/WorkloadN.java
package monorepo.workloads;

/**
 * Workload sizing shared by the bench and memory targets.
 */
public final class WorkloadN {
    private WorkloadN() {
    }

    /**
     * Workload size from WORKLOAD_N, falling back to the target's default.
     */
    public static int workloadN(int fallback) {
        String raw = System.getenv("WORKLOAD_N");
        if (raw == null) {
            return fallback;
        }
        try {
            return Integer.parseInt(raw);
        } catch (NumberFormatException e) {
            return fallback;
        }
    }
}
